cpp/include/raft/comms/comms_test.hpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
@seunghwak, that's a great point and I'm open to suggestions on this one. In short, even though the file contains test functions, I've exposed them through the public API primarily for the hand-off to the cython layer. 

I guess on one hand it could be argued that the functions are internal to RAFT so maybe they could be treated as implementation details. On the other hand, they are public to the C/C++ API of the comms package so we are trying to make explicit the points which are exposed outside of the `comms` C/C++ layer. If we were only using gtest for these functions, I think they'd just be implemented there, but these functions are also intended to enable some real-time tests for system debugging. 

Maybe it would be more explicit if we put them in a `raft/comms/test` directory? For now, it would be the only file in there unless maybe we split the current file into smaller pieces. Anyways, suggestions are welcome here.

Reply:
> but these functions are also intended to enable some real-time tests for system debugging.

If this is the case, I am not opposed to placing these files in the `include/raft(/comms)`. `include/raft/comms/test` is preferable if there are many files under this directory and we need further grouping but right at this moment, it sounds like an overkill.

If the test files are purely for internal testing, I prefer something like `include/raft_test` but I guess this is irrelevant.